File: Code/08_Threads_Mutexes/Ex6.c
#include <stdio.h>
#include <stdlib.h>
#include <pthread.h>
#include <unistd.h>

void *thread_function(void *arg);

int main()
{
	pthread_t a_thread;
	void *thread_result;
	
	printf("Criando a thread\n");
	pthread_create(&a_thread, NULL, thread_function, NULL);
	sleep(5);
	printf("Cancelando a thread\n");
	pthread_cancel(a_thread);
	printf("Thread cancelada!\n");
	return 0;
}

void *thread_function(void *arg)
{
	int i;
	printf("Nova thread\n");
	for (i = 10; i > 0; i--)
	{
		printf("Faltam %2d segundos para acabar a thread ... \n", i);
		sleep(1);
	}
	pthread_exit(0);
}

File: 2_Programação_sistemas_operacionais/2.5_Threads_Mutexes/Python/Ex6.py
import threading, time

cancel = False

def thread_function():
	global cancel
	print("Nova thread")
	for i in range(10,0,-1):
		if cancel:
			return
		print("Faltam %2d segundos para acabar a thread ..." % i)
		time.sleep(1)

print("Criando a thread")
t = threading.Thread(target=thread_function)
t.start()
time.sleep(5)
print("Cancelando a thread")
cancel = True
print("Thread cancelada!")